module/behaviour/skills/DirectWalkController/src/DirectWalkController.cpp

```diff
@@ -72,16 +79,20 @@ namespace module::behaviour::skills {
 
         on<Trigger<MotionCommand>>().then([this](const MotionCommand& command) {
             if (command.type == MotionCommand::Type::DIRECT_COMMAND) {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {26, 11}}));
-                emit(std::make_unique<WalkCommand>(subsumptionId, command.walk_command));
+                update_priority(cfg.direct_walk_priority);
+                emit(std::make_unique<WalkCommand>(subsumption_id, command.walk_command));
             }
             else if (command.type == MotionCommand::Type::STAND_STILL) {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {26, 11}}));
-                emit(std::make_unique<StopCommand>(subsumptionId));
+                update_priority(cfg.direct_walk_priority);
+                emit(std::make_unique<StopCommand>(subsumption_id));
             }
             else {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {0, 0}}));
+                update_priority(0);
             }
         });
     }
+
+    void DirectWalkController::update_priority(const float& priority) {
+        emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumption_id, {priority}}));
```

Comment:
You're only setting one priority? There are two limb sets in this action. Check the others too.